File: Examples/DotNET/CSharp/Document/SplitResources/SplitSpecificPagesToNewPDFs.cs
using System;
using System.Collections.Generic;
using Com.Aspose.Words.Api;
using Com.Aspose.Words.Model;
using Com.Aspose.Storage.Api;

namespace Document.SplitResources
{
    class SplitSpecificPagesToNewPDFs
    {
        public static void Run()
        {
            // ExStart:1
            WordsApi wordsApi = new WordsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            // Set input file name
            String fileName = "SampleWordDocument.docx";
            String format = "pdf";
            int? from = 1;
            int? to = 2;
            Boolean zipOutput = true;
            String storage = null;
            String folder = null;

            try
            {
                // Upload file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.Words Cloud SDK API to split specific pages to new pdfs
                SplitDocumentResponse apiResponse = wordsApi.PostSplitDocument(fileName, format, from, to, zipOutput, storage, folder);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    foreach (Link pageLink in apiResponse.SplitResult.Pages)
                    {
                        String outputFileName = pageLink.Href;
                        Console.WriteLine("file:" + outputFileName);
                    }
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}

File: Examples/Ruby/SplitResource/split_specific_pages_to_new_pdfs.rb
#ExStart:
require 'aspose_words_cloud'

class WorkingWithSplitResource

  include AsposeWordsCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("APP_KEY", "APP_SID")
    @words_api = WordsApi.new  
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end

  def split_specific_pages_to_new_pdfs
    file_name = "SampleWordDocument.docx"
    upload_file(file_name)

    response = @words_api.post_split_document(file_name, {format: "pdf", from: 1, to: 2, zip_output: false})
  end

end

workingWithSplitResource = WorkingWithSplitResource.new()
puts workingWithSplitResource.split_specific_pages_to_new_pdfs
#ExEnd: